api/airports/airports.controller.js

```diff
@@ -2,26 +2,47 @@ const mongoose = require('mongoose');
 const Airports = mongoose.model('airports');
 const boom = require('boom');
 
-exports.search =(req, res, next) => {
+exports.search = (req, res, next) => {
 
+    let search = req.query.data;
 
-}
+    if (search.length <= 2) {
+        next(boom.badImplementation("Please provide atleast 3 letters to search"));
+    } else {
 
+        Airports.find({name: {$regex: search,$options: 'i'}})
 
+            .then((doc) => {
+                if (doc.length === 0) {
 
+                    next(boom.badImplementation("no such airport with keyword " + search + " found"));
+                }
```

Comment:
Remove else instead use this

if (doc.length === 0) {
  return next(boom.badImplementation("no such airport with keyword " + search + " found"));
 }